opium_kernel/middlewares/error_handler.ml

```diff
@@ -0,0 +1,270 @@
+(** [Error_handler] creates error HTML pages for standard HTTP errors.
```

Comment:
I'm not entirely sure about this middleware. The change about the detailed default message looks good. But i don't think this needs to be a middleware. We already have a error handler defined at

https://github.com/rgrinberg/opium/blob/c2db89ffc317fb5d6dd2283c198b198d94e830fd/opium_kernel/server_connection.ml#L3

This can be overriden by users via an input to the Server_connection. The user needs to provide a handler with the following type

https://github.com/rgrinberg/opium/blob/c2db89ffc317fb5d6dd2283c198b198d94e830fd/opium_kernel/opium_kernel.mli#L180-L183

The httpaf runtime will watch for any exception in the handlers and wake up the error_handler when that happens. So that seems like a natural place to deal with errors that occur via an unrecoverable exception. For opium i lean towards providing a decent default httpaf error handler, (while allowing user to define their own behavior if desired), to deal with the exception scenarios. For application level errors i think it should be up to the user to create a middleware that's specific for their own application that can fill this role.

Reply:
I've pulled out the error handler from this PR and I gave a bit more context in https://github.com/rgrinberg/opium/pull/156